File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        # Sort unique values
        sorted_unique = sorted(set(arr))
        
        # Create mapping from value to rank (1-based)
        rank_map = {val: i + 1 for i, val in enumerate(sorted_unique)}
        
        # Replace each element with its rank
        return [rank_map[x] for x in arr]

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.rs
use std::collections::HashMap;

impl Solution {
    pub fn array_rank_transform(arr: Vec<i32>) -> Vec<i32> {
        // Sort unique values
        let mut sorted = arr.clone();
        sorted.sort();
        sorted.dedup(); // remove duplicates
        
        // Build mapping value -> rank (1-based)
        let mut rank_map = HashMap::new();
        for (i, &val) in sorted.iter().enumerate() {
            rank_map.insert(val, (i + 1) as i32);
        }
        
        // Replace each element
        arr.iter().map(|x| *rank_map.get(x).unwrap()).collect()
    }
}
